javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -477,6 +505,98 @@ public boolean isEmpty() {
         return TreeSet.ofAll(keyComparator(), iterator().map(Tuple2::_1));
     }
 
+    @Override
+    public Iterator<Tuple2<K, V>> descendingIterator() {
+        return entries.descendingIterator();
+    }
+
+    @Override
+    public SortedSet<K> descendingKeySet() {
+        return descendingMap().keySet();
+    }
+
+    @Override
+    public SortedMap<K, V> descendingMap() {
+        return new TreeMap<>(entries.descendingTree());
+    }
+
+    @Override
+    public SortedMap<K, V> subMap(Option<K> fromKey, boolean fromInclusive, Option<K> toKey, boolean toInclusive) {
+        return new TreeMap<>(
+                entries.subTree(
+                    fromKey.map(k -> Tuple.of(k, null)), fromInclusive,
```

Comment:
yes... and no :)

This `null` is used for search operations only as the comparator only relies on the key, and the value is not known.  That `null` is simply ignored by the comparison.

Maybe I could create a constant `Object IGNORED=null` and use it instead.